Makefile

```
@@ -53,3 +53,17 @@ test-static: static-build/install/lib/libgit2.a
 
 install-static: static-build/install/lib/libgit2.a
 	go install --tags "static" ./...
+
+define _gen_file
+	rm -fr _obj
+	go tool cgo $(1)
+	find ./_obj -type f ! -name '*.go' -exec rm {} \;
+	mv ./_obj/_cgo_gotypes.go ./_obj/cgo_gotypes.go
+	cd ./_obj && go generate ./...
+	find ./_obj -type f -name '*_string.go' -exec mv -v {} . \;
+	rm -fr ./_obj
+endef
+
+generate:
+	$(call _gen_file,diff.go)
+	$(call _gen_file,git.go)
```

is it possible to have this be

```suggestion
generate: static-build/install/lib/libgit2.a
	go generate --tags "static" ./...
```

? that way whenever a new stringer is added, the Makefile doesn't need to be updated (folks will forget about this, haha). the `_gen_file` function also depends a bit on implementation details that are not guaranteed to be true forever.

---

Unfortunately, `go generate --tags "static" ./...` will not work with CGO constants from libgit2 headers (as in `diff.go` and `git.go`), so it's a workaround to extract these constants from header files to `.go` files, see this question: https://stackoverflow.com/q/69566767/1723695

---

oh that's likely because the static library wasn't correctly built / installed (that's why the `generate:` rule has `static-build/install/lib/libgit2.a` as a dependency, which fixes this issue). i just tried this on my machine and it worked. can you give this a try?

also, minor thing, can the `generate` rule be moved to just below the `default:` one? i just realized that it's included in the various library sections, and that was a bit confusing, haha.

---

Thanks, it's working now, updated in the latest commit